parser/expression_parsing.go

```diff
@@ -59,10 +59,16 @@ func (p *Parser) parseExpression(precedence int) ast.Expression {
 		p.noPrefixParseFnError(p.curToken.Type)
 		return nil
 	}
+
 	leftExp := prefix()
 
 	for !p.peekTokenIs(token.Semicolon) && precedence < p.peekPrecedence() && p.peekTokenAtSameLine() {
+
 		infix := p.infixParseFns[p.peekToken.Type]
```

Comment:
This should be the `else` condition of the below one. It's like more a fallback now.

1. We look at next token to see if it is a call expression without paren.
2. If not, we see if it is an infix expression.
3. It's not in above conditions, just a single expression.